connect-examples/v2/python_payment/sqpaymentform.js

```diff
@@ -25,174 +20,181 @@ function requestCardNonce(event) {
+            var applePayLabel = document.getElementById('sq-apple-pay-label');
+            var masterpassBtn = document.getElementById('sq-masterpass');
+            var masterpassLabel = document.getElementById('sq-masterpass-label');
+
+            // Only show the button if Apple Pay for Web is enabled
+            // Otherwise, display the wallet not enabled message.
+            if (methods.applePay === true) {
+                applePayBtn.style.display = 'inline-block';
+                applePayLabel.style.display = 'none';
+            }
+
+            // Only show the button if Masterpass is enabled
+            // Otherwise, display the wallet not enabled message.
+            if (methods.masterpass === true) {
+                masterpassBtn.style.display = 'block';
+                masterpassLabel.style.display = 'none';
+            }
+        },
 
-        return;
-      }
+        /*
+         * callback function: createPaymentRequest
+         * Triggered when: a digital wallet payment button is clicked.
+         */
+        createPaymentRequest: function () {
+
+            return {
+                requestShippingAddress: false,
+                requestBillingInfo: true,
+                currencyCode: "USD",
+                countryCode: "US",
+                total: {
+                    label: "MERCHANT NAME",
+                    amount: "100",
+                    pending: false
+                },
+                lineItems: [
+                    {
+                        label: "Subtotal",
+                        amount: "100",
+                        pending: false
+                    }
+                ]
+            }
+        },
 
-      alert('Nonce received: ' + nonce); /* FOR TESTING ONLY */
+        /*
+         * callback function: validateShippingContact
+         * Triggered when: a shipping address is selected/changed in a digital
+         *                 wallet UI that supports address selection.
+         */
+        validateShippingContact: function (contact) {
 
-      // Assign the nonce value to the hidden form field
-      document.getElementById('card-nonce').value = nonce;
+            var validationErrorObj;
+            /* ADD CODE TO SET validationErrorObj IF ERRORS ARE FOUND */
+            return validationErrorObj;
+        },
 
-      // POST the nonce form to the payment processing page
-      document.getElementById('nonce-form').submit();
+        /*
+         * callback function: cardNonceResponseReceived
+         * Triggered when: SqPaymentForm completes a card nonce request
+         */
+        cardNonceResponseReceived: function (errors, nonce, cardData) {
+            if (errors) {
+                // Log errors from nonce generation to the Javascript console
+                console.log("Encountered errors:");
+                errors.forEach(function (error) {
+                    console.log('  ' + error.message);
+                    alert(error.message);
```

Comment:
I don't think alert for each error is a good experience. Can we make this a better error handling example?

Reply:
You're right, I'll remove the alert. I also have it outlining the text boxes in red, with a message "Please fix all errors." Should I add a special error for each one? (I would need to go back and change this on all examples as well.)